Allocation sampling garbage collection docs (#2394)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/memory-problems/index.md

![Allocation sampling.](../media/memory-problems-glitch-example-05-memory-allocation-sampling-heavy-bottom-up.msft.png)

> ## Use settings for Allocation sampling to reduce garbage

Reviewer: ```suggestion

<!-- ====================================================================== -->
## Use settings for Allocation sampling to reduce garbage
```

Reviewer: Maybe flip this sentence around so it starts with "Reduce garbage by ..."?

I also think that it might be worth explaining in a short sentence what reducing garbage really means and why it's important. Garbage itself is normal, it's the rate at which you're making the GC run that's a problem, right?

> By default, the **Allocation sampling** profiling type only reports allocations that are still alive at the end of the recording session.  Objects that are generated, garbage collected (GC'd), and then disappear aren't tracked by either **Allocation sampling** or **Allocation instrumentation on timeline**.

Reviewer: ```suggestion
By default, the **Allocation sampling** profiling type only reports allocations that are still alive at the end of the recording session.  Objects that are created, removed, and then garbage collected (GC'd) aren't displayed in the **Memory** tool when profiling using the **Allocation sampling** or **Allocation instrumentation on timeline** types.
```

You can configure the **Allocation sampling** profiling type to track objects that are discarded by GC with settings.

1. Click the **Allocation sampling** radio button.

1. Click the **Include objects discarded by major GC** and **Include objects discarded by minor GC** settings.

>     ![Allocation sampling GC settings.](../media/memory-problems-memory-allocation-sampling-gc-settings.png)

Reviewer: This image needs a red box around the two settings in question. Otherwise one has to scan the entire image to find them.

Author: @mikehoffms or @JasonAndrewWriter, can you help me with this PR by drawing red boxes on the images? For the first one, draw red boxes around the blue line when it steps down. For the second one, draw a red box around the settings under **Allocation sampling**

Reviewer 2: will chat w/ Jason re: committing such a change.  

gh rendered:
https://github.com/MicrosoftDocs/edge-developer/blob/user/zoghadya/allocation-sampling-gc-settings/microsoft-edge/devtools-guide-chromium/memory-problems/index.md#use-settings-for-allocation-sampling-to-reduce-garbage

Reviewer 3: > @mikehoffms or @JasonAndrewWriter, can you help me with this PR by drawing red boxes on the images? For the first one, draw red boxes around the blue line when it steps down. For the second one, draw a red box around the settings under **Allocation sampling**

I have it set for first thing in the morning.  Just wanted to make sure i could find the thread

Reviewer 3: I just uploaded these going to ask @zoherghadyali to take a look

>     ![Allocation sampling GC settings.](../media/memory-problems-memory-allocation-sampling-gc-settings.png)

Reviewer 2: ```suggestion
   ![Allocation sampling GC settings.](../media/memory-problems-memory-allocation-sampling-gc-settings.png)
```
make sure indents align, within any Markdown list.  Sometimes all items are pushed to the right by 1 space, when you apply 4-char indent per some linter conventions.  Can break indent or numbering if not uniform in a given list.

>     ![Allocation sampling GC settings.](../media/memory-problems-memory-allocation-sampling-gc-settings.png)

Reviewer 2: Make window as narrow as feasible while still looking recognizable/normal.  That linearly increases crispness, up to a point; eventually the resulting png is rendered narrower than the content area, so don't need to make window extremely narrow before screenshotting it.

Here is GitHub's view - different sizing than LMC, but even in this wide view, window needs to be narrower before screenshotting it.
https://github.com/MicrosoftDocs/edge-developer/blob/user/zoghadya/allocation-sampling-gc-settings/microsoft-edge/devtools-guide-chromium/memory-problems/index.md#use-settings-for-allocation-sampling-to-reduce-garbage

1. Click the **Start** button.

1. On the webpage, perform actions that you want to investigate.

1. Click the **Stop** button when you have finished all of your actions.

DevTools now tracks all of the objects that were GC'd during the recording.  Use these settings to understand how much garbage your website or app is generating.  The data reported by **Allocation sampling** will help you identify the functions that are generating the most garbage.  

> If you are investigating objects that were only GC'd during specific major or minor GC operations, configure the settings appropriately to track the operation you care about. To learn more about the differences between major and minor GC, read [Trash talk: the Orinoco garbage collector | V8 JavaScript engine developer blog](https://v8.dev/blog/trash-talk).

Reviewer: ```suggestion
If you are investigating objects that were only GC'd during specific major or minor GC operations, configure the settings appropriately to track the operation you care about. To learn more about the differences between major and minor GC, see [Trash talk: the Orinoco garbage collector | V8 JavaScript engine developer blog](https://v8.dev/blog/trash-talk).
```

Reviewer 2: 'see' is per Edge Dev Style Guide (at wiki) and is allowed, and used, by corp-level style guide:
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/term-collections/accessibility-terms

![Allocation sampling.](../media/memory-problems-glitch-example-05-memory-allocation-sampling-heavy-bottom-up.msft.png)

<!-- ====================================================================== -->

> ## Reduce garbage with additional settings for Allocation sampling

Reviewer 2: ```suggestion
## Reduce garbage with additional settings for allocation sampling
```
Judging when upper vs lowercase A:
When the context is closely focused on the UI, can lean toward "Allocation sampling".  The word 'allocation' is lowercased in the UI in mid-sentences, it's only cap'd when first word of label.

By default, the **Allocation sampling** profiling type only reports allocations that are still alive at the end of the recording session.  Objects that are created, removed, and then garbage collected (GC'd) aren't displayed in the **Memory** tool when profiling using the **Allocation sampling** or **Allocation instrumentation on timeline** types.

> You can trust the browser to clean up garbage from your code.  However, it is important to consider that GC itself is an expensive operation and multiple GCs can slow down your user's experience of your website or app.  When recording in the **Performance** tool with the **Memory** checkbox turned on, you can see the GC operation happen at the steep cliffs in the heap chart.  

Reviewer 2: ```suggestion
You can trust the browser to clean up garbage from your code.  However, it is important to consider that GC itself is an expensive operation and multiple GCs can slow down your user's experience of your website or app.  When recording in the **Performance** tool with the **Memory** checkbox turned on, you can see the GC operation happen at the steep cliffs (sudden decreases) in the heap chart.
```

> ![GC operation shown in the Performance tool.](../media/memory-problems-gc-in-performance.png)

Reviewer 2: ```suggestion
![GC operation shown in the Performance tool](../media/memory-problems-gc-in-performance.png)
```
repo is moving away from unneeded period in alt-text, for simplicity per Markdown.  not sure if i commented about that in this PR already.  might be doc'd at wiki > Edge Dev Style Guide.

To reduce the amount of garbage your code is creating and reduce the cost or the number of times GC is run, you can configure the **Allocation sampling** profiling type to track objects that are discarded by GC with settings.

> 1. Click the **Allocation sampling** radio button.

Reviewer 2: ```suggestion
1. Click the **Allocation sampling** option button.
```
check UI and check 
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/r/radio-button

1. Click the **Include objects discarded by major GC** and **Include objects discarded by minor GC** settings.

>    ![Allocation sampling GC settings.](../media/memory-problems-memory-allocation-sampling-gc-settings.png)

Reviewer 2: ```suggestion
   ![Allocation sampling GC settings](../media/memory-problems-memory-allocation-sampling-gc-settings.png)
```
